### PR TITLE
[libvirt] Add file /etc/libvirt/qemu.conf

### PR DESCRIPTION
**Pull request by zerodayz**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

**Comment by pmoravec** [comment]
Good catch.

Could you please add that file to password obfuscation in `postproc` as well? (I think you should update the loop variable from `xmlfile` to something more general)

File: sos/plugins/libvirt.py [hunk 1]
```diff
@@ -69,12 +70,17 @@ def setup(self):
...
-                self.do_file_sub(
-                    xmlfile,
-                    r"(\s*passwd=\s*')([^']*)('.*)",
-                    r"\1******\3"
-                )
+        # get files
+        config_files = [glob.glob('/etc/libvirt/qemu/*.xml'),
+                        glob.glob('/var/run/libvirt/qemu/*.xml'),
+                        glob.glob('/etc/libvirt/*.conf')]
+
```
Comment (bmr-cymru) [inline sos/plugins/libvirt.py]: I'm not keen on this: `config_files` implies a list of config files, but it's not: it's a list of lists or config  files, and then it gets flattened out in the weird looking `y for x in c for y in x`.

Why the extra complexity? Why not just assign `config_files` from the sum (append) of the three glob lists?

**Comment by zerodayz** [comment]
Is there anything else needed @bmr-cymru

**Comment by psachin** [comment]
@zerodayz I think https://github.com/sosreport/sos/pull/904/files is related and already ack'd by @battlemidget & @BryanQuigley

**Comment by zerodayz** [comment]
@psachin that one is wrong imo you are postprocessing just one file.

**Comment by bmr-cymru** [comment]
@zerodayz you're correct regarding the postprocessing, however, the commit message for your change is misleading and it really should be two separate patches:

 * One is a fix for the already-collected `/etc/libvirt/qemu/*.xml` postprocessing but this is not mentioned
 * The second adds the file `qemu.conf`

If you split these up (`git reset`/`git commit --amend`) and update the branch with `git push --force`, I'll get these merged and close the other PR.

I'm also not that keen on all the globbing and list mangling in the `postproc()` method: why not just use the `do_path_regex_sub()` method. It's designed for this type of use, supports regex syntax, and operates on the plugin's collected files list (so it doesn't go back and make more directory lookups for the globbing).

The two `.xml` rules can be done in a single call and the `.conf` one can share the same regex/replace strings via local variables.

**Comment by bmr-cymru** [comment]
I'm going to delete the 2nd hunk of this patch and re-write it:

```diff
@@ -70,10 +71,16 @@ class Libvirt(Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin):
                 self.add_copy_spec("/proc/%s/%s" % (pid, pf))
 
     def postproc(self):
-        for loc in ["/etc/", "/var/run/"]:
-            for xmlfile in glob.glob(loc + "libvirt/qemu/*.xml"):
+        # get files
+        libvirt_lists = [glob.glob('/etc/libvirt/qemu/*.xml'),
+                         glob.glob('/var/run/libvirt/qemu/*.xml'),
+                         glob.glob('/etc/libvirt/*.conf')]
+
+        # parse files
+        for libvirt_list in libvirt_lists:
+            for libvirt_file in libvirt_list:
                 self.do_file_sub(
-                    xmlfile,
+                    libvirt_file,
                     r"(\s*passwd=\s*')([^']*)('.*)",
                     r"\1******\3"
                 )
```

There is no need to iterate over a list of globs when there is already a `do_path_regex_sub()` method.

**Pull request closed without merging**